test/Common/DurableTaskEndToEndTests.cs

```diff
@@ -569,6 +569,45 @@ public async Task BatchedActorOrchestration(bool extendedSessions)
             }
         }
 
+        /// <summary>
+        /// End-to-end test which validates the wait-for-full-batch case using an actor pattern.
+        /// </summary>
+        [Theory]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task BatchedActorOrchestrationDeleteLastItemAlways(bool extendedSessions)
+        {
+            using (var host = TestHelpers.GetJobHost(
+                this.loggerProvider,
+                nameof(this.BatchedActorOrchestrationDeleteLastItemAlways),
+                extendedSessions))
+            {
+                await host.StartAsync();
+
+                var client = await host.StartOrchestratorAsync(nameof(TestOrchestrations.BatchActorRemoveLast), null, this.output);
+
+                // Perform some operations
+                await client.RaiseEventAsync("deleteItem", null); // deletes last item in the list: item5
+                await client.RaiseEventAsync("deleteItem", null); // deletes last item in the list: item4
+                await client.RaiseEventAsync("deleteItem", null); // deletes last item in the list: item3
+                await client.RaiseEventAsync("deleteItem", null); // deletes last item in the list: item2

```

Comment:
Why pass `null` for the eventData? I think you should add a `RaiseEventAsync` method to the test client which calls the new API you're introducing so that we have proper code coverage.

Reply:
The only new API we're introducing now is `WaitForExternalEvent()` (non-generic). We removed the `RaiseEventAsync()` that didn't take `eventData` because that would've been a breaking change due to the default parameter signature already in place. So, this usage is actually consistent with how it'll be used in the wild and _does_ hit the new `WaitForExternalEvent()` I've added. So we've got coverage on the change... unless you're seeing something I'm not?

Reply:
ah, gotchya. I'll revisit and push a new ver in a few.